### PR TITLE
ISSUE-21257: set redirectEnd on PerformanceResourceTiming

### LINKED ISSUE #21257: PerformanceResourceTiming: RedirectEnd
In components/net/http_loader.rs::http_fetch we need to set redirect_end.

On getting, the redirectEnd attribute MUST return as follows:

1. The time immediately after receiving the last byte of the response of the last redirect, if there are HTTP redirects or equivalent when fetching the resource and all the redirects or equivalent pass the timing allow check algorithm.
2. zero, otherwise.

https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-redirectend

### PR DESCRIPTION
**Pull request by tdelacour**
Added the appropriate attribute and setters. Included a Timer struct with a `Drop` function to set this value to zero in error cases. Set the value to `responseEnd` on success. Eliminated tests that now pass.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #21257 (GitHub issue number if applicable)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23812)

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -639,6 +637,27 @@ pub fn http_fetch(
     response
 }
 
+// Convenience struct that implements Done, for setting redirectEnd on function return
```
Comment (jdm) [inline components/net/http_loader.rs]: s/Done/Drop/?

**Review by jdm — APPROVED** [review]
Looks good! Could you squash the commits into one, please?

File: components/net/http_loader.rs (hunk 1 above)
Comment (tdelacour) [reply components/net/http_loader.rs]: lol, yes, that!

**Comment by tdelacour** [comment]
@jdm ok done!

**Comment by jdm** [comment]
@bors-servo retry
#23290

**Pull request merged**